src/System.Diagnostics.Process/tests/ProcessTests.Linux.cs

```diff
@@ -0,0 +1,86 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Security;
+using Xunit;
+using Xunit.NetCore.Extensions;
+
+namespace System.Diagnostics.Tests
+{
+    public partial class ProcessTests : ProcessTestBase
+    {
+        [Fact]
+        public void ProcessStart_UseShellExecuteTrue_OpenFile_ThrowsIfNoDefaultProgramInstalledSucceedsOtherwise()
+        {
+            string fileToOpen = GetTestFilePath() + ".txt";
+            File.WriteAllText(fileToOpen, $"{nameof(ProcessStart_UseShellExecuteTrue_OpenFile_ThrowsIfNoDefaultProgramInstalledSucceedsOtherwise)}");
+
+            string[] allowedProgramsToRun = { "xdg-open", "gnome-open", "kfmclient" };
+            foreach (var program in allowedProgramsToRun)
+            {
+                if (IsProgramInstalled(program))
+                {
+                    var startInfo = new ProcessStartInfo { UseShellExecute = true, FileName = fileToOpen };
+                    using (var px = Process.Start(startInfo))
+                    {
+                        if (px != null)
+                        {
+                            Assert.Equal(program, px.ProcessName);
+                            px.Kill();
+                            px.WaitForExit();
+                            Assert.True(px.HasExited);
+                            Assert.Equal(137, px.ExitCode); // 137 means the process was killed
+                        }
+                    }
+                    return;
+                }
+            }
+
+            Win32Exception e = Assert.Throws<Win32Exception>(() => Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = fileToOpen }));
+        }
+
+        [Fact]
+        [OuterLoop("Returns failure exit code when default program, xdg-open, is installed")]
```

Comment:
it should not fail on any of our machines. either it should succeed if xdg-open is not present, or we must require xdg-open on all our images.

Reply:
the linux image I cloned from DevTestLab doesnt have xdg-open installed by default.

@danmosemsft based on this, what would you say should be the expected behavior in this case?

do we change the images to have xdg-open installed from now on? or do I change how the API behaves so that it wont throw when xdg-open is not installed?

Reply:
Let's assume at least some of our helix (test ruN) machines have it. make the test write to console if it doesn't find it. It's outer loop so that's OK. After you've checked in, we'll see whether any of the machines actually have it. If non have it we might want to update hte images so at least some configuration is exercising the code.